### PR TITLE
make sure network path substitution matches correctly

### LINKED ISSUE #5313: Path Substitutions for NFS and libraries with multiple folders not working properly
**Describe the bug**
Path substitutions for native playback through the network are not working properly using NFS and libraries with more than one folder. The first folder on the list is correct but the second folder (and other folders) items get the wrong path.

**System (please complete the following information):**
 - OS: Linux
 - Virtualization: Docker
 - Clients: Browser, Android, Kodi
 - Browser: Edge Version 88.0.705.74 (Official build) (64-bit)
 - Jellyfin Version: 10.6.4
 - Playback: Direct Play 
 - Installed Plugins: None
 - Reverse Proxy: none
 - Base URL: none
 - Networking: Host
 - Storage: NFS

**To Reproduce**
1. Setup a library with multiple folders and (Optional) Shared network folder.
-Movie Library
-- Folder 1: /movies (Optional) Shared network folder: nfs://[IP]/Multimedia/Movies
-- Folder 2: /movies2 (Optional) Shared network folder: nfs://[IP]/Multimedia/MoviesSE
- TV Show Library
-- Folder 3: /tv (Optional) Shared network folder: nfs://[IP]/Multimedia/TVShows
-- Folder 4: /tv4 (Optional) Shared network folder: nfs://[IP]/MEDIAUSB/TVShows4
2. Scan the library
3. Go to Movies library
4. Click on the ..., then media info, on an item from the "Folder 2". Notice the path start nfs://[IP]/Multimedia/Movies2 (should be nfs://[IP]/Multimedia/MoviesSE)
5. Go to Tv Shows Library
6. Click on the ..., then media info, on an item from the "Folder 4". Notice the path start nfs://[IP]/Multimedia/TVShows4 (should be nfs://[IP]/MEDIAUSB/TVShows4)

**Expected behavior**
Expected that all folders would be substituted correctly using the (Optional) Shared network folder even if multiple folders are on the same library. 

**Additional context**
The problem only happens when you have multiple folders in the same library, and it doesn't happen on the first folder, only on 2nd, 3rd, 4th

### PR DESCRIPTION
**Pull request by cvium**
Fixes #5313

### CODE REVIEW

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/Library/PathExtensions.cs [hunk 1]
```diff
@@ -47,5 +48,84 @@ public static class PathExtensions
...
+            // when the sub path matches a similar but in-complete subpath
+            if (!subPath.EndsWith(newDirectorySeparatorChar))
+            {
+                subPath += newDirectorySeparatorChar;
+            }
+
+            if (newSubPath.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
+            {
+                newSubPath = newSubPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
+            }
```
Comment (BaronGreenback) [inline Emby.Server.Implementations/Library/PathExtensions.cs:114]: Couldn't this just be a _replace_?

```
            newSubPath = newSubPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
```

File: Emby.Server.Implementations/Library/PathExtensions.cs [hunk 2]
```diff
@@ -47,5 +48,84 @@ public static class PathExtensions
...
+
+            if (!path.Contains(subPath, StringComparison.OrdinalIgnoreCase))
+            {
+                newPath = null;
+                return false;
+            }
+
+            newPath = path.Replace(subPath, newSubPath, StringComparison.OrdinalIgnoreCase);
+            return true;
```
Comment (BaronGreenback) [inline Emby.Server.Implementations/Library/PathExtensions.cs:128]: Would this be more optimised, as I would expect _equals_ to be faster than _contains_ ???
```
            newPath = path.Replace(subPath, newSubPath, StringComparison.OrdinalIgnoreCase);
            return !string.Equals(newPath, path, StringComparison.OrdinalIgnoreCase);
```

**Review by BaronGreenback — APPROVED** [review]

File: Emby.Server.Implementations/Library/PathExtensions.cs [hunk 3]
```diff
@@ -47,5 +48,84 @@ public static class PathExtensions
...
+
+            if (string.IsNullOrWhiteSpace(subPath))
+            {
+                throw new ArgumentNullException(nameof(subPath));
+            }
+
+            if (string.IsNullOrWhiteSpace(newSubPath))
+            {
+                throw new ArgumentNullException(nameof(newSubPath));
+            }
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/PathExtensions.cs:76]: Isn't the whole point of a `Try` method that it fails gracefully?

**Review by Bond-009 — CHANGES_REQUESTED** [review]
With less checks and allocations:
(probably missed some corner cases)
```cs
diff --git a/Emby.Server.Implementations/Library/PathExtensions.cs b/Emby.Server.Implementations/Library/PathExtensions.cs
index 1fc5526ae1..ad84350171 100644
--- a/Emby.Server.Implementations/Library/PathExtensions.cs
+++ b/Emby.Server.Implementations/Library/PathExtensions.cs
@@ -1,6 +1,7 @@
 #nullable enable
 
 using System;
+using System.IO;
 using System.Diagnostics.CodeAnalysis;
 using System.Text.RegularExpressions;
 
@@ -60,19 +61,17 @@ namespace Emby.Server.Implementations.Library
         /// <exception cref="ArgumentNullException"><paramref name="path" />, <paramref name="newSubPath" /> or <paramref name="newSubPath" /> is empty.</exception>
         public static bool TryReplaceSubPath(this string path, string subPath, string newSubPath, [NotNullWhen(true)] out string? newPath)
         {
-            if (string.IsNullOrWhiteSpace(path))
-            {
-                throw new ArgumentNullException(nameof(path));
-            }
+            newPath = null;
 
-            if (string.IsNullOrWhiteSpace(subPath))
+            int subLen = subPath.Length;
+            if (path.Length == 0 || subLen == 0 || newSubPath.Length == 0)
             {
-                throw new ArgumentNullException(nameof(subPath));
+                return false;
             }
 
-            if (string.IsNullOrWhiteSpace(newSubPath))
+            if (subLen > path.Length)
             {
-                throw new ArgumentNullException(nameof(newSubPath));
+                return false;
             }
 
             char oldDirectorySeparatorChar;
@@ -91,40 +90,18 @@ namespace Emby.Server.Implementations.Library
                 newDirectorySeparatorChar = '\\';
             }
 
-            if (path.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
-            {
-                path = path.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
-            }
-
-            if (subPath.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
-            {
-                subPath = subPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
-            }
+            path = path.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
+            subPath = subPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
 
             // We have to ensure that the sub path ends with a directory separator otherwise we'll get weird results
             // when the sub path matches a similar but in-complete subpath
-            if (!subPath.EndsWith(newDirectorySeparatorChar))
-            {
-                subPath += newDirectorySeparatorChar;
-            }
-
-            if (newSubPath.Contains(oldDirectorySeparatorChar, StringComparison.Ordinal))
-            {
-                newSubPath = newSubPath.Replace(oldDirectorySeparatorChar, newDirectorySeparatorChar);
-            }
-
-            if (!newSubPath.EndsWith(newDirectorySeparatorChar))
-            {
-                newSubPath += newDirectorySeparatorChar;
-            }
-
-            if (!path.Contains(subPath, StringComparison.OrdinalIgnoreCase))
+            if (!path.StartsWith(subPath, StringComparison.OrdinalIgnoreCase)
+                || (subPath[^1] != newDirectorySeparatorChar && path[subPath.Length] != newDirectorySeparatorChar))
             {
-                newPath = null;
                 return false;
             }
 
-            newPath = path.Replace(subPath, newSubPath, StringComparison.OrdinalIgnoreCase);
+            newPath = Path.Join(newSubPath, path.AsSpan(subPath.Length));
             return true;
         }
     }
```

File: Emby.Server.Implementations/Library/LibraryManager.cs [hunk 4]
```diff
@@ -2802,22 +2796,17 @@ public string GetPathAfterNetworkSubstitution(string path, BaseItem ownerItem)
 
             if (!string.IsNullOrWhiteSpace(metadataPath) && !string.IsNullOrWhiteSpace(metadataNetworkPath))
```
Comment (Bond-009) [inline Emby.Server.Implementations/Library/LibraryManager.cs:2797]: Same

**Review by Bond-009 — CHANGES_REQUESTED** [review]
Last nitpick (I think)

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**